tests/test_basic.py

```diff
@@ -26,6 +27,47 @@ def test_basic_ping(self):
             r = os.system("ping -q 1.1.1.1 -c 1 -n > /dev/null")
             self.assertEqual(r, 0)
 
+    def test_pcap(self):
+        ''' Check -pcap capture '''
+        pcap = self.get_tmp_filename("test.pcap")
+        p = self.prun("-pcap %s" % pcap)
+        self.assertStartSync(p)
+        with self.guest_netns():
+            r = os.system("ping -q 1.1.1.1 -c 1 -n > /dev/null")
+            self.assertEqual(r, 0)
+        with open(pcap, 'rb') as f:
+            data = f.read(24)
+            header = struct.unpack(">LHHLLLL", data)
+            self.assertEqual(header[0], 0xa1b2c3d4)
+            data = f.read(16)
+            (seconds, useconds, captured_length, packet_length) = struct.unpack(">LLLL", data)
+            self.assertEqual(captured_length, 28)
+
+    def test_fd(self):
+        ''' Check -fd. '''
+        sp = socket.socketpair(type=socket.SOCK_DGRAM)
+        os.set_inheritable(sp[0].fileno(), True)
+        try :
+            p = self.prun("-fd %d" % sp[0].fileno(), close_fds=False, netns=False)
+            self.assertStartSync(p, fd=True)
+            # 10.0.2.15->10.0.2.2 ICMP Echo (ping) request
+            ping = bytes.fromhex('''
+            52 55 0a 00 02 02 70 71 aa 4b 29 aa 08 00 45 00
+            00 54 00 00 40 00 40 01 22 99 0a 00 02 0f 0a 00
+            02 02 08 00 4f 4d 73 1e 00 01 8e 3a 3b 5e 00 00
+            00 00 a7 27 06 00 00 00 00 00 10 11 12 13 14 15
+            16 17 18 19 1a 1b 1c 1d 1e 1f 20 21 22 23 24 25
+            26 27 28 29 2a 2b 2c 2d 2e 2f 30 31 32 33 34 35
+            36 37''')
+            sp[1].sendall(ping)
+            while True:
+                pong = sp[1].recv(1024)
+                if pong[14+9] == 1 and pong[14+20] == 0: #ICMP and echo reply
+                    break
+        finally:
```

Comment:
nope, don't do the try/finally block; if the test fails; sure we will leak, this is fine. if the tests fails the state of underlying resources is undefined.